src/tools/tests/recipe2plan-test.ts

```diff
@@ -219,6 +243,77 @@ $ tools/update-goldens \n\n`
       }]
     });
   }));
+  it('outputs a valid protocol buffer for resolved recipes with type restriction', Flags.withDefaultReferenceMode(async () => {
+    assert.deepEqual(
+      await protoPayloadFor(`
+        particle Writer
+          data: writes Thing {name: Text, secret: Text}
+
+        @arcId('writeArcId')
+        recipe WritingRecipe
+          thing: create 'my-handle-id' @persistent @ttl('30d')
+          Writer
+            data: writes thing
+      `), {
+      particleSpecs: [{
+        name: 'Writer',
+        isolated: false,
+        connections: [{
+          name: 'data',
+          direction: 'WRITES',
+          type: {entity: {schema: {
+            names: ['Thing'],
+            fields: {name: {primitive: 'TEXT'}, secret: {primitive: 'TEXT'}},
+            hash: 'a5856a63a2459ad6e9bcfd7ff50e23d0ed803801',
+          }}}
+        }]
+      }],
+      policies: policiesJson,
+      recipes: [{
+        name: 'WritingRecipe',
+        annotations: [{
+          name: 'arcId',
+          params: [{
+            name: 'id',
+            strValue: 'writeArcId'
+          }]
+        }],
+        handles: [{
+          fate: 'CREATE',
+          name: 'handle0',
+          id: 'my-handle-id',
+          annotations: [{
+            name: 'persistent'
+          },
+          {
+            'name': 'ttl',
+            'params': [{
+              'name': 'value',
+              'strValue': '30d'
+            }]
+          }],
+          storageKey: 'reference-mode://{db://a5856a63a2459ad6e9bcfd7ff50e23d0ed803801@arcs/Thing}{db://a5856a63a2459ad6e9bcfd7ff50e23d0ed803801@arcs/!:writeArcId/handle/my-handle-id}',
+          type: {entity: {schema: {
+            names: ['Thing'],
+            fields: {name: {primitive: 'TEXT'}},
+            hash: '25e71af4e9fc8b6958fc46a8f4b7cdf6b5f31516',
```

Comment:
here the `hash` in the storage key doesn't match the `hash` of the Handle type, because the storage key is created with the handle-connection type.

however, I'm not sure which is the right behavior. if the storage key is created with Handle type, later on policies get updated and the Handle type is less restricted - the storage key will change and the previously stored data will be lost?